x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportSemanticSearchAction.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.action;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.search.SearchRequestBuilder;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.client.internal.ParentTaskAssigningClient;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.fetch.subphase.FieldAndFormat;
+import org.elasticsearch.search.internal.SearchContext;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.ml.action.InferTrainedModelDeploymentAction;
+import org.elasticsearch.xpack.core.ml.action.SemanticSearchAction;
+import org.elasticsearch.xpack.core.ml.inference.results.TextEmbeddingResults;
+
+public class TransportSemanticSearchAction extends HandledTransportAction<SemanticSearchAction.Request, SemanticSearchAction.Response> {
+
+    private final Client client;
+    private final ClusterService clusterService;
+
+    @Inject
+    public TransportSemanticSearchAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        ClusterService clusterService
+    ) {
+        super(SemanticSearchAction.NAME, transportService, actionFilters, SemanticSearchAction.Request::new);
+        this.client = client;
+        this.clusterService = clusterService;
+    }
+
+    @Override
+    protected void doExecute(Task task, SemanticSearchAction.Request request, ActionListener<SemanticSearchAction.Response> listener) {
+
+        var parentTaskAssigningClient = new ParentTaskAssigningClient(client, clusterService.localNode(), task);
+
+        if (request.getKnnSearchBuilder() == null) {
+            listener.onFailure(
+                new IllegalArgumentException(
+                    "missing required [" + SemanticSearchAction.Request.KNN.getPreferredName() + "] section in search body"
+                )
+            );
+            return;
+        }
```

Comment:
why is this nullable at all?

Reply:
It's not nullable this check is redundant I've removed it